src/hooks/useDropzoneArea.js

```diff
@@ -0,0 +1,80 @@
+import {useState, useEffect, useCallback} from 'react';
+import {createFileFromUrl, readFile} from '../helpers';
+
+export const useDropzoneArea = ({
+    onChange,
+    clearOnUnmount,
+    initialFiles,
+    filesLimit,
+    onDelete}) => {
+    const [fileObjects, setFileObjects] = useState([]);
+
+    useEffect(() => {
```

Comment:
Some more code comments for these useEffects would be nice. Second half of file is well documented.

Reply:
Okay, I can add some comments here

Reply:
I've added some comments above each useEffect